### PR TITLE
[21.01] Raise correct exceptions in `MetadataCollection`

### PR DESCRIPTION
**Pull request by nsoranzo**
Issue found by @mvdbeek when testing https://github.com/galaxyproject/tools-iuc/blob/master/tools/jbrowse/jbrowse.xml with `planemo test`.

Introduced in commit 335a89772012041538e86835a4e16f3a2e50d51f .

Alternative to https://github.com/galaxyproject/galaxy/pull/11281 .

Also 2 other small changes that helped me during the debugging.

### CODE REVIEW

**Review by mvdbeek — DISMISSED** [review]
Awesome, thank you!

File: lib/galaxy/model/metadata.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def __getattr__(self, name):
             return self.spec[name].wrap(self.spec[name].default, object_session(self.parent))
         if name in self.parent._metadata:
             return self.parent._metadata[name]
+        raise AttributeError
```
Comment (mvdbeek) [inline lib/galaxy/model/metadata.py:116]: I think this was supposed to return `None` silently.

**Comment by mvdbeek** [comment]
Hmm, those test failure look real.

File: lib/galaxy/model/metadata.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/model/metadata.py:116]: No, see https://docs.python.org/3/reference/datamodel.html?highlight=__getattr__#object.__getattr__ : "This method should either return the (computed) attribute value or raise an `AttributeError` exception."

Comment (mvdbeek) [reply lib/galaxy/model/metadata.py:116]: But that's not how we use MetadataCollection. We're stuck with this behavior now afaict.

**Comment by nsoranzo** [comment]
> Hmm, those test failure look real.

Fixed the unit tests in a rebase, the failures were a by-product of fixing `unicodify()`, it seems that Cheetah is catching the `NotFound` exception so we need to re-raise the same exception type.

File: lib/galaxy/model/metadata.py (hunk 1 above)
Comment (nsoranzo) [reply lib/galaxy/model/metadata.py:116]: All tests pass (except for the broken kubernetes integration build), should we give it a try? Or do you have an example were this fails?

Comment (mvdbeek) [reply lib/galaxy/model/metadata.py:116]: Let's try. Thanks @nsoranzo!

Comment (mvdbeek) [reply lib/galaxy/model/metadata.py:116]: Yep, that's erroring now in sentry (https://sentry.galaxyproject.org/sentry/main/issues/1325196/):
```
AttributeError: null
  File "galaxy/web/framework/middleware/sentry.py", line 43, in __call__
    iterable = self.application(environ, start_response)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.6/site-packages/paste/recursive.py", line 85, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/middleware/statsd.py", line 33, in __call__
    req = self.application(environ, start_response)
  File "/cvmfs/main.galaxyproject.org/venv/lib/python3.6/site-packages/paste/httpexceptions.py", line 640, in __call__
    return self.application(environ, start_response)
  File "galaxy/web/framework/base.py", line 136, in __call__
    return self.handle_request(environ, start_response)
  File "galaxy/web/framework/base.py", line 215, in handle_request
    body = method(trans, **kwargs)
  File "galaxy/webapps/galaxy/controllers/visualization.py", line 488, in display_by_username_and_slug
    visualization_config = self.get_visualization_config(trans, visualization)
  File "galaxy/webapps/base/controller.py", line 1035, in get_visualization_config
    tracks.append(pack_track(drawable_dict))
  File "galaxy/webapps/base/controller.py", line 996, in pack_track
    "filters": track_dict.get('filters', {'filters' : track_data_provider.get_filters()}),
  File "galaxy/visualization/data_providers/genome.py", line 249, in get_filters
    if self.original_dataset.metadata.viz_filter_cols:
  File "galaxy/model/metadata.py", line 116, in __getattr__
    raise AttributeError
```
That's code we have control over, so we could catch the attribute error here, but there's probably tool code does such checks that should just return None.

**Pull request merged**